networkx/algorithms/dag.py

```diff
@@ -287,3 +290,94 @@ def is_aperiodic(G):
         return g==1
     else:
         return g==1 and nx.is_aperiodic(G.subgraph(set(G)-set(levels)))
+
+@not_implemented_for('undirected')
+def transitive_closure(G):
+    """ Returns transitive closure of a directed graph
+
+    The transitive closure of G = (V,E) is a graph G+ = (V,E+) such that
+    for all v,w in V there is an edge (v,w) in E+ if and only if there
+    is a non-null path from v to w in G.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    TC : NetworkX DiGraph
+       Graph
+
+    Raises
+    ------
+    NetworkXNotImplemented
+      If G is not directed
+
+    References
+    ----------
+    .. [1] http://www.ics.uci.edu/~eppstein/PADS/PartialOrder.py
+
+    """
+    TC = nx.DiGraph()
+    TC.add_nodes_from(G.nodes_iter())
+    TC.add_edges_from(G.edges_iter())
+    for v in G:
+        TC.add_edges_from((v, u) for u in nx.dfs_preorder_nodes(G, source=v)
+                          if v != u)
+    return TC
+
+@not_implemented_for('undirected')
+def antichains(G):
+    """Generates antichains from a DAG.
+
+    An antichain is a subset of a partially ordered set such that any
+    two elements in the subset are incomparable.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    antichain : generator object
+
+    Raises
+    ------
+    NetworkXNotImplemented
+      If G is not directed
+
+    NetworkXUnfeasible
+       If G contains a cycle
+
+    Notes
+    -----
+    This function was originally developed by Peter Jipsen and Franco Saliola
+    for the SAGE project. It's included in NetworkX with permission from the
+    authors. Original SAGE code at:
+
+    https://sage.informatik.uni-goettingen.de/src/combinat/posets/hasse_diagram.py
+
+    References
+    ----------
+    .. [1] Free Lattices, by R. Freese, J. Jezek and J. B. Nation,
+    AMS, Vol 42, 1995, p. 226.
+
+    """
+    # nx.topological_sort(G) will raise NetworkXUnfeasible error
+    # if there is a cycle in G.
+    nx.topological_sort(G)
+    TC = nx.transitive_closure(G)
+    antichains_stacks = [([], nx.topological_sort(G, reverse=True))]
```

Comment:
Sorry for not having been clear: here is where we need to call `nx.topological_sort` (and then `NetworkXUnfeasible` can be raised if G contains a cycle). No need to call it two lines before (and without assigning it to any variable!). I'd like to keep this function close to the one that wrote the original authors from SAGE, which is quite beatiful. So you should also remove the comment about raising `NetworkXUnfeasible`, this is already in the docstrings.